app/src/main/java/com/findmyrecycling/MainActivity.kt

```diff
@@ -29,84 +30,94 @@ class MainActivity : ComponentActivity() {
-                .fillMaxWidth()
-                .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
-                .border(width = 1.dp,
+                        shape = RoundedCornerShape(4.dp)
+                    ),
+            )
+
+            OutlinedTextField(
+                value = location,
+                onValueChange = { location = it },
+                label = { Text(stringResource(R.string.location)) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
+                    .border(
+                        width = 1.dp,
                         color = Color.Black,
-                        shape = RoundedCornerShape(4.dp)),
-        )
+                        shape = RoundedCornerShape(4.dp)
+                    ),
+            )
 
 
+        }
     }
-}
 
-@Preview(showBackground = true)
-@Composable
-fun DefaultPreview() {
-    FindMyRecyclingTheme {
-        RecycleSearch("Android")
+    @Preview(showBackground = true)
+    @Composable
+    fun DefaultPreview() {
+        FindMyRecyclingTheme {
+            RecycleSearch("Android")
+        }
+
+        Button(
+            onClick = {
+                signIn()
+            }
+        ) {
+            Text(text = "Logon")
+        }
+    }
+
+    private fun signIn() {
+        val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
+        val signInIntent = AuthUI.getInstance()
+            .createSignInIntentBuilder()
+            .setAvailableProviders(providers)
+            .build()
+
+        signInLauncher.launch(signInIntent)
     }
 
-    Button(
-        onClick = {
-            signIn()
+    private val signInLauncher = registerForActivityResult(
+        FirebaseAuthUIActivityResultContract()
+    ) { res ->
+        this.signInResult(res)
+    }
+
+
+    private fun signInResult(result: FirebaseAuthUIAuthenticationResult) {
+        val response = result.idpResponse
+        if (result.resultCode == RESULT_OK) {
+            var user = FirebaseAuth.getInstance().currentUser
+        } else {
+            Log.e("MainActivity.kt", "Error logging in " + response?.error?.errorCode)
         }
-    ) {
-        Text(text = "Logon")
     }
-}
-
-fun signIn() {
-    val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
-    val signinIntent = AuthUI.getInstance()
-        .createSignInIntentBuilder()
-        .setAvailableProviders(providers)
-        .build()
-
-    signInLauncher.launch(signinIntent)
-}
-
-private val signInLauncher = registerForActivityResult (
-    FirebaseAuthUIActivityResultContract()
-) {
-        res -> this.signInResult(res)
-}
-
-
-private fun signInResult(result: FirebaseAuthUIAuthenticationResult) {
-    val response = result.idpResponse
-    if (result.resultCode == ComponentActivity.RESULT_OK) {
-        var user = FirebaseAuth.getInstance().currentUser
-    }
\ No newline at end of file
+}
```

Comment:
Git thinks I changed the whole file but I only removed/replaced some brackets so the program would build without errors.